### ISSUE TITLE
Recalculate Totals button ignores quantities

### ISSUE THREAD

**Issue by pippinsplugins** [easydigitaldownloads/Easy-Digital-Downloads#4245]
When adding new items to payment records, the `Recalculate Totals` button is shown and automatically updates the total amount on the payment record based on the item amounts and quantities.

If an item is added with a quantity greater than one, only the amount of a single item is included in the recalculation.

This should have recalculated as $68 but since it ignored the 2x on the $24 item, the total came out to $44.
![screen shot 2016-01-27 at 8 52 57 am](https://cloud.githubusercontent.com/assets/1034109/12592968/a718bf7e-c4d3-11e5-9765-a2c0661e42db.png)

**Comment by pippinsplugins** [comment]
Ah, small nuance. The quantities are only ignored on freshly added items that have been added but not saved. After the payment has been saved, recalculating includes quantities as expected.

**Comment by cklosowski** [comment]
:+1: thanks for that info @pippinsplugins

**Comment by cklosowski** [comment]
This seems to be working fine for me on master:
![screen shot 2016-01-26 at 7 39 22 pm](https://cloud.githubusercontent.com/assets/1390124/12602203/a2d1a5da-c464-11e5-94ad-94883176a1a4.png)

![screen shot 2016-01-26 at 7 39 39 pm](https://cloud.githubusercontent.com/assets/1390124/12602207/ac19093a-c464-11e5-8f07-b603cdb82c94.png)

![screen shot 2016-01-26 at 7 39 44 pm](https://cloud.githubusercontent.com/assets/1390124/12602210/affc3950-c464-11e5-9344-31fa44708a52.png)

**Comment by cklosowski** [comment]
I think the issue is actually in the display. If I actually add items it works correctly and the reacalculation works, but as soon as I save...the line items change:
Before hitting update:
![screen shot 2016-01-26 at 7 44 26 pm](https://cloud.githubusercontent.com/assets/1390124/12602291/44a0cf30-c465-11e5-884a-2d481cda6bf0.png)

![screen shot 2016-01-26 at 7 44 28 pm](https://cloud.githubusercontent.com/assets/1390124/12602292/47733e96-c465-11e5-827f-d24ce0ec005f.png)

After hitting update:
![screen shot 2016-01-26 at 7 44 40 pm](https://cloud.githubusercontent.com/assets/1390124/12602297/4be6f724-c465-11e5-81e0-cd2af7175555.png)

**Comment by pippinsplugins** [comment]
PR almost fixes it. I found one case that fails: adding a second copy of the same download.

The payment started with just the top Love It Pro then I added the second one with a custom amount:
![screen shot 2016-01-28 at 9 02 59 am](https://cloud.githubusercontent.com/assets/1034109/12648114/ff31e490-c59d-11e5-92a9-11034445958e.png)

After saving the payment, the sales log for the second Love It Pro was not created.

I then tested it again by adding a different download and that worked fine.

Recalculation was correct in both places.

I'm pretty sure this is not a new bug so I'm okay with not resolving it in this issue. Thoughts @cklosowski ?
